app/controllers/tags_controller.rb

```diff
@@ -0,0 +1,43 @@
+class TagsController < ApplicationController
+  before_action :authenticate_user!
+  before_action -> { check_permission('admin') }
```

Comment:
i'm about to move my `#show` furniture into this controller. on the assumption that an anon user should be able to look at a tag homepage, what do i do with the above authentication logic to make sure it doesn't block non-logged-in users from accessing the `#show` action? is this already the case by default? (i'm more worried about line 3 than 2)

Reply:
we can change it to when we merge
``` rb
before_action :authenticate_user!, except: [:show]
before_action -> { check_permission('admin') }, except [:show]
```

Reply:
👍